core-kotlin-modules/core-kotlin-concurrency-4/src/test/kotlin/com/baeldung/continuation/ContinuationLiveTest.kt

```diff
@@ -0,0 +1,127 @@
+            doWork("Job 2", 600)
+        }
+
+        val job3 = launch(dispatcher) {
+            doWork("Job 3", 100)
+        }
+
+        job1.join()
+        job2.join()
+        job3.join()
+
+        logger.info("All coroutines finished!")
+    }
+
+    @Test
+    fun `test create continuation manually`(): Unit = runBlocking {
+        val simpleContinuation = Continuation<Int>(Dispatchers.IO) { result ->
+            assertEquals(45, result.getOrNull())
+        }
+        simpleContinuation.resumeWith(Result.success(45))
+    }
+
+    @Test
+    fun `test continuation using suspendCoroutine`() = runBlocking {
+        val result = suspendCoroutine { continuation ->
+            continuation.resumeWith(Result.success("Baeldung"))
+        }
+        assertEquals("Baeldung", result)
+    }
+
+    // using resumeWith()
+    private suspend fun simulateNetworkRequestResume(url: String): String {
+        return withContext(Dispatchers.IO) {
+            suspendCoroutine { continuation ->
+                val result = try {
+                    val connection = (URL(url).openConnection() as HttpURLConnection).apply {
+                        requestMethod = "GET"
+                        connectTimeout = 5000
+                        readTimeout = 5000
+                    }
+
+                    val responseCode = connection.responseCode
+                    if (responseCode == HttpURLConnection.HTTP_OK) {
+                        Result.success("$responseCode")
+                    } else {
+                        Result.failure(Exception("$responseCode - Failed"))
+                    }
+                } catch (e: Exception) {
+                    Result.failure(Exception("${e.message} - Failed"))
+                }
+
+                continuation.resumeWith(result)
+            }
+        }
+    }
+
+    // using resume() and resumeWithException()
+    private suspend fun simulateNetworkRequest(url: String): String {
+        return withContext(Dispatchers.IO) {
+            suspendCoroutine { continuation ->
+                try {
+                    val connection = (URL(url).openConnection() as HttpURLConnection).apply {
+                        requestMethod = "GET"
+                        connectTimeout = 5000
+                        readTimeout = 5000
+                    }
+
+                    val responseCode = connection.responseCode
+                    if (responseCode == HttpURLConnection.HTTP_OK) {
+                        continuation.resume("$responseCode")
+                    } else {
+                        continuation.resumeWithException(Exception("HTTP response code $responseCode - Failed"))
+                    }
+                } catch (e: Exception) {
+                    continuation.resumeWithException(Exception("${e.message} - Failed"))
+                }
+            }
+        }
+    }
+
+    @Test
+    fun `test continuation using suspendFunction network call`(): Unit = runBlocking {
```

Comment:
This test provides very minimal test coverage - most of the branches of the methods under test remain untested